embassy-nrf/src/pwm.rs

```diff
@@ -376,20 +418,36 @@ pub struct SequenceConfig {
 impl Default for SequenceConfig {
     fn default() -> SequenceConfig {
         SequenceConfig {
-            counter_mode: CounterMode::Up,
-            max_duty: 1000,
-            prescaler: Prescaler::Div16,
-            sequence_load: SequenceLoad::Common,
             refresh: 0,
             end_delay: 0,
         }
     }
 }
 
+/// A composition of a sequence buffer and its configuration.
+#[non_exhaustive]
+pub struct Sequence<'d> {
+    /// The words comprising the sequence. Must not exceed 32767 words.
+    pub words: &'d mut [u16],
```

Comment:
Why must this buffer be mutable? It's not required by the PWM as far as I can tell? If you need to write again to the buffer, you could instead let the Sequence go out of scope in your app, which should allow you to write to the buf and create a new Sequence instance. That way feels more immutable.

Reply:
Thanks for feedback.

You can’t let the sequence go out of scope because the PWM is using it until you tell it to stop. The PWM needs to own the buffer between starting and stopping. 

Hopefully the example I provide shows why we need the mutability. When we stop the PWM we get the buffer refs back so we can then update the contents inline. In my example, we lighten and darken the colours. 

I hope I’m not missing something…

Reply:
I can see the convenience argument. What I was thinking of is that instead of 
```  
        if let (Some(color_seq), Some(reset_seq)) = pwm.stop() {
            color_seq.words[color_bit] = bit_value;
            unwrap!(pwm.start(color_seq, Some(reset_seq), SequenceMode::Times(2)));
        }
```

You could do

```  
        pwm.stop();
        color_seq_words[color_bit] = bit_value;
        let color_seq = Sequence::new(&color_seq_words, SequenceConfig::default());
        let reset_seq = Sequence::new(&reset_seq_words, reset_seq_config);
        unwrap!(pwm.start(color_seq, Some(reset_seq), SequenceMode::Times(2)));
```

I.e. recreating the Sequence instances with the same buffer references. But I can see how that's a bit more work. I've not explored if it can be done more elegantly.

Reply:
But you can’t subsequently modify color_seq_words once you pass it into the start method... That buffer must live until the Pwm struct is dropped or the stop method is called… it has to be owned by the Pwm struct… 

Thanks again though. If you could try modifying the example then that’d be great. I could be missing something for sure.

Reply:
Right, and the pwm.stop() in the above code will allow you to modify color_seq_words again, because the PWM would no longer own it. But FWIW I think maybe your approach is easier to use, so I'm fine with merging this as is.

Reply:
The sequence buffer is no longer mutable. A new Sequences struct can easily be dropped to return ownership of the Pwm driver and then establish new sequences.